packages/cli/fern-definition/validator/src/rules/valid-generic/__test__/valid-generic.test.ts

```diff
@@ -0,0 +1,80 @@
+import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
+import { getViolationsForRule } from "../../../testing-utils/getViolationsForRule";
+import { ValidGenericRule } from "../valid-generic";
+
+describe("valid-generic", () => {
+    it("simple", async () => {
+        const violations = await getViolationsForRule({
+            rule: ValidGenericRule,
+            absolutePathToWorkspace: join(
+                AbsoluteFilePath.of(__dirname),
+                RelativeFilePath.of("fixtures"),
+                RelativeFilePath.of("simple")
+            )
+        });
+
+        expect(violations).toEqual([
+            {
+                severity: "error",
+                relativeFilepath: "1.yml",
+                nodePath: ["types", "GenericApplicationObject"],
+                message: 'Generic values in object properties are not supported: "foo: GenericUsedType<string>".'
+            },
+            {
+                severity: "error",
+                relativeFilepath: "1.yml",
+                nodePath: ["types", "GenericApplicationDiscriminatedUnion"],
```

Comment:
ideally this is `types, GenericApplicationDiscriminatedUnion, foo` is there any way to make that happen?